### PR TITLE
feat(docker): add arm64 build

### PR DESCRIPTION
**Pull request by danielkelemen**
related to camunda/camunda-bpm-platform#2668

### CODE REVIEW

**Comment by mleiter06** [comment]
watching!

**Review by bstncartwright — APPROVED** [review]

File: release.sh [hunk 1]
```diff
@@ -4,13 +4,27 @@ EE=${EE:-false}
...
-    docker push ${IMAGE}:${tag}
+function build_and_push {
+    local tags=("$@")
+    printf -v tag_arguments -- "--tag $IMAGE:%s " "${tags[@]}"
+    docker buildx build .                         \
+        $tag_arguments                            \
+        --build-arg DISTRO=${DISTRO}              \
+        --build-arg EE=${EE}                      \
+        --build-arg USER=${NEXUS_USER}            \
+        --build-arg PASSWORD=${NEXUS_PASS}        \
```
Comment (tasso94) [inline release.sh:21]: Can you check if the credentials are retrievable via docker history? If yes, that would be problematic.
Cf. https://stackoverflow.com/a/68187363

File: pipeline.sh [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
     --build-arg EE=${EE}                      \
     --build-arg USER=${NEXUS_USER}            \
-    --build-arg PASSWORD=${NEXUS_PASS}
+    --build-arg PASSWORD=${NEXUS_PASS}        \
+    --cache-to type=gha,scope="$GITHUB_REF_NAME-$DISTRO-image" \
+    --cache-from type=gha,scope="$GITHUB_REF_NAME-$DISTRO-image" \
+    --load
 
-./test/test.sh
+docker inspect "${IMAGE_NAME}" | grep "Architecture" -A2
```
Comment (tasso94) [inline pipeline.sh:17]: Sweet.

File: .github/workflows/build-test-and-publish-ce.yml [hunk 3]
```diff
@@ -6,18 +6,55 @@ jobs:
...
+        uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        if: ${{ matrix.PLATFORM != 'amd64' }}
+        with:
+            platforms: ${{ matrix.PLATFORM }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Expose GitHub Runtime
+        uses: crazy-max/ghaction-github-runtime@v2
```
Comment (tasso94) [inline .github/workflows/build-test-and-publish-ce.yml]: Do we really want to add a dependency to `crazy-max/ghaction-github-runtime@v2` that has only 19 stars and does something relatively simple? Can we just pass the GitHub Runtime environment variables we need to the build/release step? This makes also more transparent where things come from.

**Review by tasso94 — CHANGES_REQUESTED** [review]
Looks good to me in general. I added some hints/questions below.

**Comment by tasso94** [comment]
When I search for the JRE `openjdk11-jre-headless` [1] that is used inside the image in the Alpine package registry, I cannot find an ARM variant of it. How does this actually work? If it's emulated, should we move to an ARM-native JRE instead? Can you also check the other packages?
https://github.com/camunda/docker-camunda-bpm-platform/blob/ff7f677a0ab9d280893a03fee3507e19c5055693/Dockerfile#L61-L68

[1] https://pkgs.alpinelinux.org/packages?name=openjdk11-jre-headless&branch=v3.15&repo=&arch=&maintainer=

**Edit:** Found it. `aarch64` is effectively ARM64.

**Comment by danielkelemen** [comment]
> When I search for the JRE `openjdk11-jre-headless` [1] that is used inside the image in the Alpine package registry, I cannot find an ARM variant of it. How does this actually work? If it's emulated, should we move to an ARM-native JRE instead? Can you also check the other packages?

That's the [aarch64](https://en.wikipedia.org/wiki/AArch64) package:
```sh
/ # apk add openjdk11-jre-headless
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz
(1/8) Installing java-common (0.5-r0)
...
```
I guess it's also the one installed for the arm/v6 and arm/v7 packages because alpine has those variants too.
Update: Nope, for those you cannot install the `openjdk11-jre-headless` package.

File: .github/workflows/build-test-and-publish-ce.yml (hunk 3 above)
Comment (danielkelemen) [reply .github/workflows/build-test-and-publish-ce.yml]: It's crazy-max! 😄
I'll refactor it...

File: release.sh (hunk 1 above)
Comment (danielkelemen) [reply release.sh:21]: Link to the same question: https://github.com/camunda/jenkins-job-dsl-seed-jobs/pull/380#discussion_r1064768499

**Review by tasso94 — APPROVED** [review]

**Pull request merged**